packages/video_player/video_player/android/src/main/java/io/flutter/plugins/videoplayer/VideoPlayerPlugin.java

```diff
@@ -85,109 +85,86 @@ private void onDestroy() {
-
-          VideoPlayer player;
-          if (call.argument("asset") != null) {
-            String assetLookupKey;
-            if (call.argument("package") != null) {
-              assetLookupKey =
-                  flutterState.keyForAssetAndPackageName.get(
-                      call.argument("asset"), call.argument("package"));
-            } else {
-              assetLookupKey = flutterState.keyForAsset.get(call.argument("asset"));
-            }
-            player =
-                new VideoPlayer(
-                    flutterState.applicationContext,
-                    eventChannel,
-                    handle,
-                    "asset:///" + assetLookupKey,
-                    result,
-                    null);
-            videoPlayers.put(handle.id(), player);
-          } else {
-            player =
-                new VideoPlayer(
-                    flutterState.applicationContext,
-                    eventChannel,
-                    handle,
-                    call.argument("uri"),
-                    result,
-                    call.argument("formatHint"));
-            videoPlayers.put(handle.id(), player);
-          }
-          break;
-        }
-      default:
-        {
-          long textureId = ((Number) call.argument("textureId")).longValue();
-          VideoPlayer player = videoPlayers.get(textureId);
-          if (player == null) {
-            result.error(
-                "Unknown textureId",
-                "No video player associated with texture id " + textureId,
-                null);
-            return;
-          }
-          onMethodCall(call, result, textureId, player);
-          break;
-        }
-    }
+  public VoidMessage init(VoidMessage arg) {
+    disposeAllPlayers();
+    return new VoidMessage();
   }
 
-  private void onMethodCall(MethodCall call, Result result, long textureId, VideoPlayer player) {
-    switch (call.method) {
-      case "setLooping":
-        player.setLooping(call.argument("looping"));
-        result.success(null);
-        break;
-      case "setVolume":
-        player.setVolume(call.argument("volume"));
-        result.success(null);
-        break;
-      case "play":
-        player.play();
-        result.success(null);
-        break;
-      case "pause":
-        player.pause();
-        result.success(null);
-        break;
-      case "seekTo":
-        int location = ((Number) call.argument("location")).intValue();
-        player.seekTo(location);
-        result.success(null);
-        break;
-      case "position":
-        result.success(player.getPosition());
-        player.sendBufferingUpdate();
-        break;
-      case "dispose":
-        player.dispose();
-        videoPlayers.remove(textureId);
-        result.success(null);
-        break;
-      default:
-        result.notImplemented();
-        break;
+  TextureMessage create(CreateMessage arg) {
+    TextureRegistry.SurfaceTextureEntry handle =
```

Comment:
I wish GitHub was fancy enough to have critique style detect-moves. I'll just assume these are simple moves.